File: src/main/java/random/telegramhomebot/telegram/BotProperties.java
package random.telegramhomebot.telegram;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Profile;
import random.telegramhomebot.config.ProfileService;

import java.util.List;

@Data
@Profile("!" + ProfileService.MOCK_BOT)
@Configuration
@ConfigurationProperties(prefix = "telegram")
public class BotProperties {
    private String apiUrl;
    private Long botOwnerId;
    private String token;
    private String botName;
    private List<Long> homeGroupUserIds;
}

File: src/main/java/random/telegramhomebot/telegram/BotProperties.kt
package random.telegramhomebot.telegram

import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component

@Component
@ConditionalOnProperty(prefix = "mock-bot", value = ["enabled"], havingValue = "false")
@ConfigurationProperties(prefix = "telegram")
class BotProperties {
    lateinit var apiUrl: String
    lateinit var botOwnerId: Number
    lateinit var token: String
    lateinit var botName: String
    lateinit var homeGroupUserIds: List<Long>
}
